src/pages/MyTransfers/TableFilters.js

```diff
@@ -0,0 +1,150 @@
+ * @description Renders the date range filter
+ *
+ * @return {JSX.Element} Date range filter component
+ * @constructor
+ */
+export const DateRangeFilter = () => {
+  // get filter from context
+  const { filter, setFilter } = useTransfersContext();
+  // const { after, before } = filter;
+
+  // start and end dates (for datepicker display)
+  // they are dayjs date objects
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const defaultDateText = 'mm/dd/yyyy';
+  // MM/DD/YYYY format is used to convert date object to text
+  const dateFormat = defaultDateText.toUpperCase();
+
+  // opens the popover
+  const handleClick = (e) => {
+    setAnchorEl(e.target);
+  };
+
+  // resets dates to previously present values and closes the popover
+  const handleClose = () => {
+    setStartDate(null);
+    setEndDate(null);
+    setAnchorEl(null);
+  };
+
+  // updates the filter object and closes the popver
+  const handleOK = () => {
+    const newFilter = new TransferFilter({ ...filter, after: startDate, before: endDate });
+    setFilter(newFilter);
+
+    setAnchorEl(null);
+  };
+
+  return (
+    <FormControl sx={{ width: '192px' }} data-testid='date-range-filter'>
+      <FilterLabelText>Created Date</FilterLabelText>
+      <DateRangeButton onClick={handleClick} endIcon={<DateRangeFilterIcon />}>
+        {startDate ? getDateText(startDate, dateFormat) : defaultDateText} - {endDate ? getDateText(endDate, dateFormat) : defaultDateText}
+      </DateRangeButton>
+      <Popover
+        open={!!anchorEl}
+        anchorEl={anchorEl}
+        onClose={handleClose}
+        anchorOrigin={{
+          vertical: 'top',
+          horizontal: 'left',
+        }}>
+        <Card sx={{ maxWidth: '250px' }}>
+          <CardContent>
+            <Grid container rowGap={'18px'}>
+              <DateFilterTitle item xs={12}>
+                <Typography sx={{ fontSize: '24px', fontWeight: '700' }}>Enter dates</Typography>
+                <DateRangeIcon sx={{ fontSize: '24px' }} />
+              </DateFilterTitle>
+              <Grid item xs={12}>
+                <DatePicker
+                  label={'Start date'}
+                  value={startDate}
+                  onChange={(date) => setStartDate(date)}
+                  disableFuture
+                  maxDate={endDate}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <DatePicker
+                  label={'End date'}
+                  value={endDate}
+                  onChange={(date) => setEndDate(date)}
+                  disableFuture
+                  minDate={startDate}
+                />
+              </Grid>
+            </Grid>
+          </CardContent>
+          <CardActions sx={{ justifyContent: 'flex-end' }}>
+            <Button onClick={handleClose}>Cancel</Button>
+            <Button onClick={handleOK}>OK</Button>
```

Comment:
It should not be possible to submit the form when there are errors in day controls: "ok" button should be disabled. Otherwise, I will see error on the page. 
![dates](https://github.com/Greenstand/treetracker-wallet-admin-client/assets/19413810/fbc69845-1d1d-45a2-adb0-25b6ba17e131)